contentcuration/contentcuration/frontend/channelEdit/vuex/clipboard/actions.js

```diff
@@ -89,76 +90,150 @@ export function loadClipboardNodes(context, { parent, ancestorId }) {
+      const selectionParent = isRootInsert ? node.source_channel_id : node.parent;
+
+      // Determine parent selection state so we can add new descendant nodes
+      // with the appropriate selection state
+      const parentSelection = context.getters.currentSelectionState(selectionParent);
+
+      context.commit('UPDATE_SELECTION_STATE', {
+        id: node.id,
+        selectionState:
+          parentSelection & SelectionFlags.ALL_DESCENDANTS ? SELECT_ALL : SelectionFlags.NONE,
+      });
+    }
+  });
+
+  context.commit('ADD_CLIPBOARD_NODES', nodes);
+  return nodes;
+}
+
+/**
+ * Dedicated Vuex action for loading nodes from the IndexedDB listener
+ */
+export function addClipboardNodeFromListener(context, obj) {
+  let promise = Promise.resolve();
+
+  // Load channel and create color if not already present on the clipboard
+  if (!context.getters.channelIds.includes(obj.source_channel_id)) {
+    promise = context.dispatch('loadChannels', { id__in: [obj.source_channel_id] });
+  }
+
+  return promise
+    .then(() =>
+      context.dispatch(
+        'contentNode/loadContentNodes',
+        { '[node_id+channel_id]__in': [obj.source_node_id, obj.source_channel_id] },
+        { root }
+      )
+    )
+    .then(() =>
+      context.dispatch('addClipboardNodes', {
+        nodes: [obj],
+        parent: obj.parent,
+      })
+    )
+    .then(() => context.dispatch('loadChannelColors'));
+}
+
 let preloadPromise = Promise.resolve();
 /**
  * Queues request to load the clipboard nodes, and returns a promise
  * when this particular set is loaded
  *
  * @param context
- * @param {{ parent: String, [ancestorId]: String|null }} payload
+ * @param {{ parent: String }} payload
  * @return {Promise<Boolean>}
  */
-export function preloadClipboardNodes(context, payload) {
-  const key = preloadKey(payload);
-  context.commit('ADD_PRELOAD_NODES', payload);
-
+export function preloadClipboardNodes(context, { parent }) {
+  context.commit('SET_PRELOAD_NODES', {
+    [parent]: LoadStatus.NOT_LOADED,
+  });
   preloadPromise = preloadPromise.then(() => {
-    if (key in context.state.preloadNodes) {
-      return context
-        .dispatch('cancelPreloadClipboardNodes', payload)
-        .then(() => context.dispatch('loadClipboardNodes', payload))
-        .then(() => new Promise(resolve => defer(resolve)))
-        .then(() => true);
-    }
-    return false;
+    return context
+      .dispatch('doPreloadClipboardNodes', { parent })
+      .then(() => new Promise(resolve => defer(resolve)));
   });
   return preloadPromise;
 }
 
+export function doPreloadClipboardNodes(context, { parent }) {
+  if (context.state.preloadNodes[parent] === LoadStatus.NOT_LOADED) {
+    context.commit('SET_PRELOAD_NODES', {
+      [parent]: LoadStatus.PRELOADING,
+    });
+    return context.dispatch('loadClipboardNodes', { parent }).then(() =>
+      context.commit('SET_PRELOAD_NODES', {
+        [parent]: LoadStatus.LOADED,
+      })
+    );
+  }
+}
```

Comment:
Preloading was simplified without need for `ancestorId`, and also added a status constant for clearer handling.